purchase_order_line_seller_used/models/purchase_order_line.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2023 ForgeFlow S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    seller_used_id = fields.Many2one(
+        comodel_name="product.supplierinfo", readonly=True, copy=False
+    )
+
+    @api.onchange("product_qty", "product_uom")
+    def _onchange_quantity(self):
+        """
+        IMPORTANT: We search the Vendor Pricelist just as done in Odoo standard
+        """
+        result = super()._onchange_quantity()
+        if not self.product_id or self.invoice_lines:
+            return result
+        params = {"order_id": self.order_id}
+        seller = self.product_id._select_seller(
+            partner_id=self.partner_id,
+            quantity=self.product_qty,
+            date=self.order_id.date_order and self.order_id.date_order.date(),
+            uom_id=self.product_uom,
+            params=params,
+        )
+        if seller:
+            self.seller_used_id = seller
```

Comment:
If seller is not found, seller_used_id should be set to False, dont you think ?

Reply:
I think it makes more sense, I will change. Thank you 😄